nemo/collections/common/data/lhotse/cutset.py

```diff
@@ -47,6 +53,184 @@ def read_cutset_from_config(config) -> Tuple[CutSet, bool]:
+        There are two levels of weights: per task (outer) and per dataset (inner).
+        The final weight is the product of outer and inner weight::
+        input_cfg:
+          - type: group
+            weight: 0.7
+            tags:
+              task: asr
+            input_cfg:
+              - type: nemo_tarred
+                manifest_filepath: /path/to/asr1/manifest__OP_0..512_CL_.json
+                tarred_audio_filepath: /path/to/tarred_audio/asr1/audio__OP_0..512_CL_.tar
+                weight: 0.6
+                tags:
+                  lang: en
+                  some_metadata: some_value
+              - type: nemo_tarred
+                manifest_filepath: /path/to/asr2/manifest__OP_0..512_CL_.json
+                tarred_audio_filepath: /path/to/asr2/tarred_audio/audio__OP_0..512_CL_.tar
+                weight: 0.4
+                tags:
+                  lang: pl
+                  some_metadata: some_value
+          - type: group
+            weight: 0.3
+            tags:
+              task: ast
+            input_cfg:
+              - type: nemo_tarred
+                manifest_filepath: /path/to/ast1/manifest__OP_0..512_CL_.json
+                tarred_audio_filepath: /path/to/ast1/tarred_audio/audio__OP_0..512_CL_.tar
+                weight: 0.2
+                tags:
+                  src_lang: en
+                  tgt_lang: pl
+              - type: nemo_tarred
+                manifest_filepath: /path/to/ast2/manifest__OP_0..512_CL_.json
+                tarred_audio_filepath: /path/to/ast2/tarred_audio/audio__OP_0..512_CL_.tar
+                weight: 0.8
+                tags:
+                  src_lang: pl
+                  tgt_lang: en
+    """
+    propagate_attrs = {
+        "shuffle": config.shuffle,
+        "shard_seed": config.shard_seed,
+        "text_field": config.text_field,
+        "lang_field": config.lang_field,
+        "missing_sampling_rate_ok": config.missing_sampling_rate_ok,
+        "max_open_streams": config.max_open_streams,
+    }
+    cuts, is_tarred = parse_and_combine_datasets(config.input_cfg, propagate_attrs=propagate_attrs)
+    return cuts, is_tarred
+
+
+def parse_group(grp_cfg: DictConfig, propagate_attrs: dict) -> [CutSet, bool]:
+    assert grp_cfg.type in KNOWN_DATASET_CONFIG_TYPES, f"Unknown item type in dataset config list: {grp_cfg.type=}"
+    if grp_cfg.type == "nemo_tarred":
+        is_tarred = True
+        cuts = read_nemo_manifest(grp_cfg, is_tarred=is_tarred)
+    elif grp_cfg.type == "nemo":
+        is_tarred = False
+        cuts = read_nemo_manifest(grp_cfg, is_tarred=is_tarred)
+    elif grp_cfg.type == "lhotse_shar":
+        is_tarred = True
+        cuts = read_lhotse_manifest(grp_cfg, is_tarred=is_tarred)
+    elif grp_cfg.type == "lhotse":
+        is_tarred = False
+        cuts = read_lhotse_manifest(grp_cfg, is_tarred=is_tarred)
+    elif grp_cfg.type == "txt":
+        is_tarred = True
```

Comment:
good catch; it's in order to make them treated "as if" they were tarred, which will create a replica of the sampler object in each dataloading worker, moving text reading to the dataloader workers. Technically it could work with this set to False too (minus a bit of efficiency), but this code doesn't support mixing tarred and non-tarred datasets, and I imagine we want to mostly mix text + audio in a tarred audio setting. I'll add a relevant comment